hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/multipart/S3MultipartUploadCompleteRequest.java

```diff
@@ -546,22 +476,6 @@ protected String getDBOzoneKey(OMMetadataManager omMetadataManager,
     return omMetadataManager.getOzoneKey(volumeName, bucketName, keyName);
   }
 
-  protected void addMissingParentsToCache(OmBucketInfo omBucketInfo,
-      List<OmDirectoryInfo> missingParentInfos,
-      OMMetadataManager omMetadataManager,
-      long volumeId, long bucketId, long transactionLogIndex
-  ) throws IOException {
-    // FSO is disabled. Do nothing.
-  }
-
-  protected void addMultiPartToCache(
-      OMMetadataManager omMetadataManager, String multipartOpenKey,
-      OMFileRequest.OMPathInfoWithFSO pathInfoFSO, OmKeyInfo omKeyInfo,
-      String keyName, long transactionLogIndex
-  ) throws IOException {
-    // FSO is disabled. Do nothing.
-  }
```

Comment:
This used to do nothing for non-FSO, and was overridden to add the missing parents only for FSO.  I may be missing something, but it seems the patch does not distinguish between these two cases.

Reply:
Addressed. Seems for all the S3 request, they pair with a FSO request. My new change only apply this logic to FSO requests. There is one request which does not pair with FSO is the `S3ExpiredMultipartUploadsAbortRequest`. I applied logic to this request as well since my issue is this request.